pkg/utils/handler.go

```diff
@@ -164,14 +257,18 @@ func (h *Handler) CheckAgents() ([]string, []string, error) {
 	}
 	for _, pod := range pods.Items {
 		agentName := pod.ObjectMeta.Name
-		agentData, exists := h.AgentCache[agentName]
-		if !exists {
+		agent, err := h.ExtensionsClientset.Agents().Get(agentName)
```

Comment:
Is there a method of getting all the data using one request? Maybe List() is suitable for this?